### PR TITLE
Render pass device

### LINKED ISSUE #14: feat(RenderTask): Add CUDA support
Noticed that this is slightly different from, and mandatory for #3 to be useful.

### PR DESCRIPTION
**Pull request by nkemnitz**
Resolves #14, probably useful for #3

### CODE REVIEW

File: corgie/cli/merge_render.py [hunk 1]
```diff
@@ -97,6 +97,7 @@ def __init__(self, src_layers, src_specs, dst_layer, mip, pad, bcube):
         self.bcube = bcube
 
     def execute(self):
+        device = 'cuda' if torch.cuda.is_available() else 'cpu'
```
Comment (nkemnitz) [inline corgie/cli/merge_render.py:100]: I guess this should be part of the task - related to #3

File: corgie/layers/base.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ def read(self, dtype=None, **kwargs):
         # TODO: if np type is unit32, convert it to int64
         if data_np.dtype == np.uint32:
             data_np = data_np.astype(np.int64)
-        data_tens = torch.as_tensor(data_np, device=self.device)
+        data_tens = torch.as_tensor(data_np, device=kwargs.get('device', None))
```
Comment (nkemnitz) [inline corgie/layers/base.py:43]: Added here because `BaseLayerType.__init__()` is never called in the worker?

**Comment by supersergiy** [comment]
hm, I think `BaseLayerType.__init__()` should be called whenever a `VolumetricLayer` is created. If it doesn't, it's some sort of bug.

Here's why I think it should be called:
`VolumetricLayer` inherits from `BaseLayerType`, then calls `super` [here](https://github.com/seung-lab/corgie/blob/5421a392c2417998121fa726194a2f040e585c67/corgie/layers/volumetric_layers.py#L22).

Could you explain a bit why do you think it's not getting called?

**Comment by nkemnitz** [comment]
>Could you explain a bit why do you think it's not getting called?

I added a `corgie_logger.debug` and a pdb call to the init function for some tasks queued in SQS. The workers completely ignored it...
Oh, maybe it has something to do with how tasks and layers are pickled and unpickled, instead of recreated(?)

**Comment by nkemnitz** [comment]
Ah, found it: https://docs.python.org/3.6/library/pickle.html#pickling-class-instances

>When a class instance is unpickled, its `__init__()` method is usually _not_ invoked.

**Comment by supersergiy** [comment]
Got it. Another reason to do away with pickle.

Let's adapt your approach for now, and I'll try to remove the pickle from the code base by the end of next week. This would make the tasks smaller, task serialization faster, and the workflow more intuitive.

**Pull request merged**